test/suite/test_compact07.py

```diff
@@ -0,0 +1,88 @@
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import time
+import wiredtiger, wttest
+
+# test_compact07.py
+# Test background compaction interruption.
+class test_compact07(wttest.WiredTigerTestCase):
+    uri = 'file:test_compact07'
+    # Make compact slow to increase the chances of interruption. 
+    conn_config = 'timing_stress_for_test=[compact_slow]'
+    create_params = 'key_format=i,value_format=S,allocation_size=4KB,leaf_page_max=32KB,'
+    # Have enough tables to give the server something to work on.
+    num_tables = 5
+    table_numkv = 100 * 1000
+    table_uri='table:test_compact07'
+
+    delete_range_len = 10 * 1000
+    delete_ranges_count = 4
+    value_size = 1024 # The value should be small enough so that we don't create overflow pages.
+
+    # Create the table in a way that it creates a mostly empty file.
+    def test_compact07(self):
+
+        # FIXME-WT-11399
+        if self.runningHook('tiered'):
+            self.skipTest("this test does not yet work with tiered storage")
+
+        # Create the tables and populate them.
+        for i in range(0, self.num_tables):
+            uri = f'{self.table_uri}_{i}'
+            self.session.create(uri, self.create_params)
+            c = self.session.open_cursor(uri, None)
+            for k in range(self.table_numkv):
+                c[k] = ('%07d' % k) + '_' + 'abcd' * ((self.value_size // 4) - 2)
+            c.close()
+        self.session.checkpoint()
+
+        # Now let's delete a lot of data ranges. Create enough space so that compact runs in more
+        # than one iteration.
+        for i in range(0, self.num_tables):
+            uri = f'{self.table_uri}_{i}'
+            c = self.session.open_cursor(uri, None)
+            for r in range(self.delete_ranges_count):
+                start = r * self.table_numkv // self.delete_ranges_count
+                for i in range(self.delete_range_len):
+                    c.set_key(start + i)
+                    c.remove()
+            c.close()
+
+        # Make sure compaction will start working by setting a low threshold.
+        compact_cfg_on = 'background=true,free_space_target=1MB'
+        compact_cfg_off = 'background=false'
+        with self.expectedStderrPattern('background compact interrupted by application'):
+            self.session.compact(None, compact_cfg_on)
+            # Give the background compaction server some time to wake up.
+            time.sleep(1)
+            # Interrupt it.
+            self.session.compact(None, compact_cfg_off)
```

Comment:
We should check the background compact server status stats after this.

Reply:
Good idea, checking this in e898ddfbc6b65a1d4a2e6f5185905846035fde7c.